Index: gdm3-2.30.5/daemon/gdm-xdmcp-display.c
===================================================================
--- gdm3-2.30.5.orig/daemon/gdm-xdmcp-display.c	2010-08-11 16:19:44.000000000 +0200
+++ gdm3-2.30.5/daemon/gdm-xdmcp-display.c	2010-10-18 17:20:41.000000000 +0200
@@ -41,6 +41,10 @@
 #include "gdm-common.h"
 #include "gdm-address.h"
 
+#include "gdm-settings.h"
+#include "gdm-settings-direct.h"
+#include "gdm-settings-keys.h"
+
 #define GDM_XDMCP_DISPLAY_GET_PRIVATE(o) (G_TYPE_INSTANCE_GET_PRIVATE ((o), GDM_TYPE_XDMCP_DISPLAY, GdmXdmcpDisplayPrivate))
 
 struct GdmXdmcpDisplayPrivate
@@ -190,6 +194,14 @@
         *enabledp = FALSE;
         *usernamep = g_strdup ("");
         *delayp = 0;
+        gboolean allow_remote_autologin;
+
+        allow_remote_autologin = FALSE;
+        gdm_settings_direct_get_boolean (GDM_KEY_ALLOW_REMOTE_AUTOLOGIN, &allow_remote_autologin);
+
+        if ( allow_remote_autologin ) {
+                GDM_DISPLAY_CLASS (gdm_xdmcp_display_parent_class)->get_timed_login_details (display, enabledp, usernamep, delayp);
+        }
 }
 
 static void
Index: gdm3-2.30.5/data/gdm.schemas.in.in
===================================================================
--- gdm3-2.30.5.orig/data/gdm.schemas.in.in	2010-10-18 17:20:40.000000000 +0200
+++ gdm3-2.30.5/data/gdm.schemas.in.in	2010-10-18 17:20:41.000000000 +0200
@@ -79,6 +79,11 @@
       <signature>b</signature>
       <default>true</default>
     </schema>
+    <schema>
+      <key>security/AllowRemoteAutoLogin</key>
+      <signature>b</signature>
+      <default>false</default>
+    </schema>
 
     <schema>
       <key>greeter/Include</key>
Index: gdm3-2.30.5/common/gdm-settings-keys.h
===================================================================
--- gdm3-2.30.5.orig/common/gdm-settings-keys.h	2010-10-18 17:20:40.000000000 +0200
+++ gdm3-2.30.5/common/gdm-settings-keys.h	2010-10-18 17:20:41.000000000 +0200
@@ -44,6 +44,7 @@
 #define GDM_KEY_INCLUDE_ALL "greeter/IncludeAll"
 
 #define GDM_KEY_DISALLOW_TCP "security/DisallowTCP"
+#define GDM_KEY_ALLOW_REMOTE_AUTOLOGIN "security/AllowRemoteAutoLogin"
 
 #define GDM_KEY_XDMCP_ENABLE "xdmcp/Enable"
 #define GDM_KEY_MAX_PENDING "xdmcp/MaxPending"
