Index: gdm3-2.30.2/daemon/gdm-xdmcp-display-factory.c
===================================================================
--- gdm3-2.30.2.orig/daemon/gdm-xdmcp-display-factory.c	2010-06-02 19:07:43.000000000 +0200
+++ gdm3-2.30.2/daemon/gdm-xdmcp-display-factory.c	2010-06-02 19:07:43.000000000 +0200
@@ -2887,6 +2887,8 @@
                 return TRUE;
         }
 
+        ss_len = (int)gdm_sockaddr_len (&clnt_ss);
+
         res = XdmcpFill (factory->priv->socket_fd, &factory->priv->buf, (XdmcpNetaddr)&clnt_ss, &ss_len);
         if G_UNLIKELY (! res) {
                 g_debug ("GdmXdmcpDisplayFactory: Could not create XDMCP buffer!");
